server/public/model/file_info.go

```diff
@@ -121,7 +121,7 @@ func (fi *FileInfo) IsValid() *AppError {
 }
 
 func (fi *FileInfo) IsImage() bool {
-	return strings.HasPrefix(fi.MimeType, "image")
+	return strings.HasPrefix(fi.MimeType, "image") || strings.EqualFold(fi.Extension, "heic") || strings.EqualFold(fi.Extension, "heif")
```

Comment:
so the mime type of heif file is `image/heif` and that of heic file is `image/heic`, can we use those instead of checking for file extensions? File extensions are unreliable at best.

Reply:
This is what happens if I incorrectly name a file as heic extension, so we should rely on mime type.
![Screenshot 2024-06-06 at 9 54 49 AM](https://github.com/mattermost/mattermost/assets/18575143/76d5e284-8864-4dd1-8ce4-d37cb1f195c0)

Reply:
+1 on this one. We shouldn't rely on file extensions.